modules/docs/docgen/spec/componentParser.spec.ts

```diff
@@ -0,0 +1,388 @@
+      expect(docs).toHaveProperty('0.type.displayName', 'B');
+
+    });
+
+    it('should handle a default prop from JSDoc', () => {
+      const program = createProgramFromSource('test.tsx',`
+        import React from 'react'
+
+        export class A extends React.Component<Props> {}
+
+        interface Props {
+          /** @default {'b'} */
+          a: string
+        }
+      `);
+
+      const docs = parse(program, 'test.tsx', [componentParser]);
+
+      expect(docs).toHaveProperty('0.name', 'A');
+      expect(docs).toHaveProperty('0.type.kind', 'component');
+      expect(docs).toHaveProperty('0.type.props.0.kind', 'property');
+      expect(docs).toHaveProperty('0.type.props.0.name', 'a');
+      expect(docs).toHaveProperty('0.type.props.0.type.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.props.0.type.value', 'string');
+      expect(docs).toHaveProperty('0.type.props.0.defaultValue.kind', 'string');
+      expect(docs).toHaveProperty('0.type.props.0.defaultValue.value', 'b');
+    });
+
+    it('should handle a default prop from defaultProps', () => {
+      const program = createProgramFromSource('test.tsx',`
+        import React from 'react'
+
+        export class A extends React.Component<Props> {
+          static defaultProps = {
+            a: 'b'
+          }
+        }
+
+        interface Props {
+          a: string
+        }
+      `);
+
+      const docs = parse(program, 'test.tsx', [componentParser]);
+
+      expect(docs).toHaveProperty('0.name', 'A');
+      expect(docs).toHaveProperty('0.type.kind', 'component');
+      expect(docs).toHaveProperty('0.type.props.0.kind', 'property');
+      expect(docs).toHaveProperty('0.type.props.0.name', 'a');
+      expect(docs).toHaveProperty('0.type.props.0.type.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.props.0.type.value', 'string');
+      expect(docs).toHaveProperty('0.type.props.0.defaultValue.kind', 'string');
+      expect(docs).toHaveProperty('0.type.props.0.defaultValue.value', 'b');
+    });
+  })
+
+  describe('function components', () => {
+    it('should handle a function that returns JSX', () => {
+      const program = createProgramFromSource('test.tsx',`
+        import React from 'react'
+
+        export function A (props: Props) {
+          return <div />
+        }
+
+        interface Props {
+          a: string
+        }
+      `);
+
+      const docs = parse(program, 'test.tsx', [componentParser]);
+
+      expect(docs).toHaveProperty('0.name', 'A');
+      expect(docs).toHaveProperty('0.type.kind', 'component');
+      expect(docs).toHaveProperty('0.type.props.0.kind', 'property');
+      expect(docs).toHaveProperty('0.type.props.0.name', 'a');
+      expect(docs).toHaveProperty('0.type.props.0.type.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.props.0.type.value', 'string');
+    })
+
+    it('should handle deconstruction defaults a function that returns JSX', () => {
```

Comment:
This reads weird, and the next spec has the same description.

Reply:
Good catch. This spec is verifying the `@default` JSDoc tag. I'll update.